staging/src/k8s.io/apiserver/pkg/admission/interfaces.go

```diff
@@ -49,6 +50,8 @@ type Attributes interface {
 	GetKind() schema.GroupVersionKind
 	// GetUserInfo is information about the requesting user
 	GetUserInfo() user.Info
+	// GetAuditEvent returns audit event of the request
+	GetAuditEvent() *audit.Event
```

Comment:
I think a much more scoped change to this interface is needed (not exposing the full audit event)

The same comments about delegation, serialization, and ownership of the object apply here. Every attribute in `admission.Attributes` must be able to be serialized and sent to a mutating webhook admission plugin. How would an audit.Event be sent remotely?

The things an in-tree admission plugin can do should also be able to be done by a mutating webhook admission plugin. How would a remote admission webhook contribute audit annotations?

Reply:
Thanks for the suggestion, I will investigate it.

Reply:
Would it make sense to just add a `map[string]string` return value to the admit interface? Ditto for authorization & authentication interfaces?

Reply:
I think the question is how to expeand this interface gracefully:

https://github.com/kubernetes/kubernetes/blob/1789a8513c745f274ff8ddc3c46948a4a2ff2468/staging/src/k8s.io/api/admission/v1beta1/types.go#L40